### ISSUE TITLE
Try to give more useful error when launch fails due to misformed template.

### ISSUE THREAD

**Issue by cmillion** [MillionConcepts/killscreen#3]
The error in this case currently looks like:
```
File ~/github/hostess/hostess/aws/ec2.py:565, in Cluster.launch(cls, count, template, options, tags, client, session, wait, *instance_args, **instance_kwargs)
    563 for _ in range(5):
    564     try:
--> 565         instances = instance_hook()
    566         break
    567     except ClientError as ce:

File ~/github/hostess/hostess/aws/ec2.py:556, in Cluster.launch.<locals>.instance_hook()
    554 def instance_hook():
    555     return instances_from_ids(
--> 556         fleet["Instances"][0]["InstanceIds"],
    557         *instance_args,
    558         client=client,
    559         **instance_kwargs,
    560     )

IndexError: list index out of range
```

Examples of reasons this failed (because of a misconstructed template) were because the instance type was not specified or because gpu or spot resources were given tag values in an instance that used neither.

It could be more meaningful by parsing / returning the error message returned by, e.g.

```
from hostess.aws.utilities import init_client
client = init_client('ec2')
client.describe_instances()
client.create_fleet(
            LaunchTemplateConfigs=[
                {
                    "LaunchTemplateSpecification": {
                        "LaunchTemplateName": 'a_template',
                        "Version": "$Default",
                    }
                }
            ],
            TargetCapacitySpecification={
                "TotalTargetCapacity": 1,
                "OnDemandTargetCapacity": 1,
                "DefaultTargetCapacityType": "on-demand",
            },
            Type="instant",
        )
```

There are probably other things that will cause `IndexError` at `fleet["Instances"][0]["InstanceIds"]` so maybe this would need to be way more complicated to not be misleading. 🤷

**Comment by m-stclair** [comment]
it does in fact have to be way more complicated to not be misleading. you have to parse the error message returned by the EC2 API.  what _would not_ be very complicated would be extracting the error message and returning it to the user, although many of them are somewhat opaque.

**Comment by cmillion** [comment]
But less opaque than just the `IndexError` in at least _some_ cases.

**Comment by m-stclair** [comment]
very true! easy enough to check for errors in the API response and populate an exception from them if present. implemented in 0e9b8f2. closing this for now.
